src/main/java/core/mate/academy/service/BulldozerProducer.java

```diff
@@ -0,0 +1,22 @@
+package core.mate.academy.service;
+
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BulldozerProducer implements MachineProducer {
+    private List<? extends Machine> bulldozers = new ArrayList<>();
+    private MachineService<Bulldozer> bulldozerService = new MachineServiceImpl();
+
+    @Override
+    public List<Bulldozer> get() {
+        Bulldozer bulldozer = new Bulldozer();
+        bulldozer.setPower(1000);
+        bulldozer.setName("Bulldozer");
+        bulldozer.setColor("Blue");
+
+        bulldozerService.fill((List<? super Machine>) bulldozers, bulldozer);

```

Comment:
no need to use bulldozerService here

just instantiate the list of bulldozers, add with a new bulldozer and then return it.